src/idl_gen_go.cpp

```diff
@@ -784,7 +844,7 @@ class GoGenerator : public BaseGenerator {
   }
 
   std::string GenTypeGet(const Type &type) {
-    if (type.enum_def != nullptr && !type.enum_def->is_union) {
```

Comment:
Can you talk me through why you removed this branch of the if statement?

Reply:
This is responsible for this change in generated code:
```
-func (rcv *Monster) TestType() byte {
+func (rcv *Monster) TestType() Any {
```
Enums with `is_union` set are the autogenerated "_type" fields for unions, as I understand it. Following other enum fields, it seems like it should return the actual enum type instead of `byte`, for better type checking, etc. It works with the change on line 407, to use `GenTypePointer` in `GetUnionField`. (Since `obj` is an out-argument, `GenTypePointer` makes more sense to me there anyway.)

Actually, this part of the change could have been made earlier, since the enum type was an alias for `byte`. Was there a reason for using `byte` explicitly?

Reply:
SGTM